helm-charts/common/nginx/values.yaml

```diff
@@ -0,0 +1,113 @@
+  # Specifies whether a service account should be created
+  create: false
+  # Automatically mount a ServiceAccount's API credentials?
+  automount: true
+  # Annotations to add to the service account
+  annotations: {}
+  # The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name: ""
+
+# This is for setting Kubernetes Annotations to a Pod.
+# For more information checkout: https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
+podAnnotations: {}
+# This is for setting Kubernetes Labels to a Pod.
+# For more information checkout: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/
+podLabels: {}
+
+podSecurityContext: {}
+  # fsGroup: 2000
+
+securityContext: {}
+  # readOnlyRootFilesystem: true
+  # allowPrivilegeEscalation: false
+  # runAsNonRoot: true
+  # runAsUser: 1000
+  # capabilities:
+  # drop:
+  #  - ALL
+  # seccompProfile:
+  #  type: RuntimeDefault
+
+# This is for setting up a service more information can be found here: https://kubernetes.io/docs/concepts/services-networking/service/
+service:
+  # This sets the service type more information can be found here: https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types
+  type: ClusterIP
+  # This sets the ports more information can be found here: https://kubernetes.io/docs/concepts/services-networking/service/#field-spec-ports
+  port: 80
+
+resources: {}
+  # We usually recommend not to specify default resources and to leave this as a conscious
+  # choice for the user. This also increases chances charts run on environments with little
+  # resources, such as Minikube. If you do want to specify resources, uncomment the following
+  # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
+  # limits:
+  #   cpu: 100m
+  #   memory: 128Mi
+  # requests:
+  #   cpu: 100m
+  #   memory: 128Mi
+
+# This section is for setting up autoscaling more information can be found here: https://kubernetes.io/docs/concepts/workloads/autoscaling/
+# Enabling HPA will:
+# - Ignore above replicaCount, as it will be controlled by HPA
+# - Add example HPA scaling rules with thresholds suitable for Xeon deployments
+# - Require custom metrics ConfigMap available in the main application chart
+autoscaling:
+  enabled: false
+  minReplicas: 1
+  maxReplicas: 4
+  # targetCPUUtilizationPercentage: 80
+  # targetMemoryUtilizationPercentage: 80
+
+# Additional volumes on the output Deployment definition.
+volumes: []
+#  - name: tmp
+#    emptyDir: {}
+
+# Additional volumeMounts on the output Deployment definition.
+volumeMounts: []
+#  - name: tmp
+#    mountPath: /tmp
+
+nodeSelector: {}
+
+tolerations: []
+
+affinity: {}
+
+# Name of the ConfigMap to be used as the environment variables
+# See https://github.com/opea-project/GenAIComps/blob/main/comps/nginx/nginx.conf.template for details
+cmName: ""
```

Comment:
Applications' hard-code NGINX `configMap` name to  `{{ .Release.Name }}-nginx-config`.

So I do not see a point in having this configurable.  Just hard-coded it in NGINX `Deployment` to a name applications will use, like all the other configMaps for common components do.

Reply:
The reason I leave this configurable is to make CI happy. Otherwise, a hardcoded configmap would prevent the nginx's own CI test failure due to unable to schedule pod, because of this configmap is defined in the application e2e chart, not in the nginx chart itself.

Reply:
done